src/fpm_os.F90

```diff
@@ -0,0 +1,79 @@
+module fpm_os
+    use, intrinsic :: iso_c_binding, only : c_char, c_int, c_null_char
+    use fpm_error, only : error_t, fatal_error
+    implicit none
+    private
+    public :: change_directory, get_current_directory
+
+#ifndef _WIN32
+    character(len=*), parameter :: pwd_env = "PWD"
+#else
+    character(len=*), parameter :: pwd_env = "CD"
+#endif
+
+    interface
+        function chdir(path) result(stat) &
+#ifndef _WIN32
+                bind(C, name="chdir")
+#else
+                bind(C, name="_chdir")
+#endif
```

Comment:
It's fine for this PR, but down the road we should move all these platform specific ifdefs into a `platform.f90` or `compatibility.f90` file, and then just `use` such a file here.

Reply:
I was tempted to do:
```f90
#ifndef _WIN32
character(len=*), parameter :: chdir_symbol = "chdir"
#else
character(len=*), parameter :: chdir_symbol = "_chdir"
#endif
! ...
function chdir(...) bind(C, name=chdir_symbol)
```
But GFortran doesn't like those, unfortunately.

Reply:
I was going to try exactly that to see if it works! Then we could have the `chdir_symbol` in `platform.f90`. As it is now, we'll have to move the whole `chdir` definition into `platform.f90`, which is fine too.

Reply:
I used string concatenation in the past with `bind(C, name=namespace//"...")` so in principle this seems to work. But I'm hitting:

```
./src/fpm_os.F90:18:29:

   18 |                 bind(C, name=chdir_symbol)
      |                             1
Error: Parameter ‘chdir_symbol’ at (1) has not been declared or is a variable, which does not reduce to a constant expression
```

Reply:
Maybe this is not possible because the interface is a separate scoping unit not automatically inheriting the module scope. Using `import` will also not work here because it would make the parameter available after it has been used. That's a tricky one, maybe worth a thread on discourse?

Reply:
Posted here: Posted here: https://fortran-lang.discourse.group/t/constant-expressions-for-bind-c-name/1319

Let's see if somebody more knowledgeable in the Fortran standard can help us out here.